torch/utils/data/datapipes/_typing.py

```diff
@@ -339,10 +339,44 @@ def _hash_(self):
 
 class _IterDataPipeMeta(_DataPipeMeta):
     r"""
-    Metaclass for `IterDataPipe` and inherits from `_DataPipeMeta`. Aad a hook function to `__iter__`.
+    Metaclass for `IterDataPipe` and inherits from `_DataPipeMeta`. Aad various functions for behaviors
+    specific to `IterDataPipe`.
     """
 
     def __new__(cls, name, bases, namespace, **kwargs):
+
+        if 'reset' in namespace:
```

Comment:
I think `reset` is always in the namespace.

Reply:
Actually it is not always in the namespace. It is only in the namespace if the class has the method `reset` defined, it doesn't seem like it is inheriting from `IterDataPipe`'s namespace at this point.

Reply:
#TIL